cdcv2/metadata/sql/tables.go

```diff
@@ -97,10 +185,26 @@ func (*ScheduleDO) TableName() string {
 
 // ProgressDO mapped from table <progress>
 type ProgressDO struct {
-	CaptureID string           `gorm:"column:capture_id;type:varchar(128);primaryKey" json:"capture_id"`
-	Progress  *CaptureProgress `gorm:"column:progress;type:longtext" json:"progress"`
-	Version   uint64           `gorm:"column:version;type:bigint(20) unsigned;not null" json:"version"`
-	UpdateAt  time.Time        `gorm:"column:update_at;type:datetime(6);not null;autoUpdateTime" json:"update_at"`
+	CaptureID model.CaptureID           `gorm:"column:capture_id;type:varchar(128);primaryKey" json:"capture_id"`
+	Progress  *metadata.CaptureProgress `gorm:"column:progress;type:longtext" json:"progress"`
+
+	Version  uint64    `gorm:"column:version;type:bigint(20) unsigned;not null" json:"version"`
+	UpdateAt time.Time `gorm:"column:update_at;type:datetime(6);not null;autoUpdateTime" json:"update_at"`
+}
+
+// GetKey returns the key of the progress.
+func (p *ProgressDO) GetKey() model.CaptureID {
+	return p.CaptureID
+}
+
+// GetVersion returns the version of the progress.
+func (p *ProgressDO) GetVersion() uint64 {
```

Comment:
why do we need these functions? it's Java-style?

Reply:
We need the interface to implement a generic based cache layer. Ref: https://github.com/pingcap/tiflow/pull/9792/files#diff-df5be604641b0a21457ede9555f4e6e1a3cc86cd74e2759ff48ca0d9700a0739R132-R136